src/classes/Player.js

```diff
@@ -132,6 +124,42 @@ class Player {
     }
   }
 
+  async createStream() {
+    const url = queue[this.songEntry];
+    if (url.includes('youtube.com')) {
+      const stream = this.createYoutubeStream()
+      this.song = info.title;
+
+      stream.once(streamEvents.info, ({ title }) => {
+        this.song = title;
+        if (!this.updateDispatcherStatus()) {
+          this.updateSongPresence();
+        }
+      });
+
+    } else if (url.includes('soundcloud.com') && !!this.soundcloudClientID) {
+      const stream = await this.createSoundcloudStream();
+      const info = await scdl.getInfo(url, this.soundcloudClientID);
+
+      this.song = info.title;
+      if (!this.updateDispatcherStatus()) {
+        this.updateSongPresence();
+      }
+      return stream
```

Comment:
Same as for the youtube one, move this inside the `createSoundcloudStream()` method.

This leaves `createStream()` as a method that will select the corresponding stream without seeing how each one is set up.